lightclient/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod rpc_to_internal;
+
+extern crate core;
+
+use anyhow::Result;
+use filecoin_f3_blssig::BLSVerifier;
+use filecoin_f3_certs as certs;
+use filecoin_f3_certs::validate_finality_certificates;
+use filecoin_f3_gpbft::{ECChain, PowerEntries};
+use filecoin_f3_rpc::RPCClient;
+
+pub struct LightClient {
+    rpc: RPCClient,
+    network_name: String,
```

Comment:
I recommend passing an enum here. It's more extensible.

See https://github.com/ChainSafe/forest-explorer/blob/97f4618776dbe93a480a4216aa453c60025a13c8/src/faucet/constants.rs#L66-L71